src/main/java/net/dancier/dancer/chat/ChatController.java

```diff
@@ -0,0 +1,87 @@
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static net.dancier.dancer.authentication.Constants.ROLE_USER;
+
+@RestController
+@RequestMapping("/chats")
+@RequiredArgsConstructor
+public class ChatController {
+    private final static Logger log = LoggerFactory.getLogger(ContactController.class);
+
+    private final ChatService chatService;
+
+    @GetMapping("")
+    @Secured(ROLE_USER)
+    public ResponseEntity<ChatsDto> getChats(@CurrentUser AuthenticatedUser authenticatedUser) {
+        log.info("Fetching chats for user {}.", authenticatedUser.getUserId());
+        return ResponseEntity.ok(
+                chatService.getChatsByUserId(authenticatedUser.getUserId())
+        );
+    }
+
+    @PostMapping("")
+    @Secured(ROLE_USER)
+    public ResponseEntity<ChatDto> postChat(
+            @CurrentUser AuthenticatedUser authenticatedUser,
+            @RequestBody CreateChatDto createChatDto) {
+        log.info("Creating a new chat for user {}.", authenticatedUser.getUserId());
+        return new ResponseEntity(
+                chatService.createChat(authenticatedUser.getUserId(), createChatDto), HttpStatus.CREATED
+        );
+    }
+
+    @GetMapping("/{chatId}")
+    @Secured(ROLE_USER)
+    public ResponseEntity<ChatDto> getChat(
+            @CurrentUser AuthenticatedUser authenticatedUser,
+            @PathVariable UUID chatId) {
+        log.info("Fetching single chat {} for user {}.", chatId, authenticatedUser.getUserId());
+        return ResponseEntity.ok(
+                chatService.getChat(chatId, authenticatedUser.getUserId())
+        );
+    }
+
+    @GetMapping("/{chatId}/messages")
+    @Secured(ROLE_USER)
+    public ResponseEntity<MessagesDto> getMessages(
+            @CurrentUser AuthenticatedUser authenticatedUser,
+            @PathVariable UUID chatId,
+            @RequestParam Optional<UUID> lastMessageId) {
+        log.info("Fetching messages for chat {} for user {}.", chatId, authenticatedUser.getUserId());
+        return ResponseEntity.ok(
+                chatService.getMessages(chatId, authenticatedUser.getUserId(), lastMessageId)
+        );
+    }
+
+    @PostMapping("/{chatId}/messages")
+    @Secured(ROLE_USER)
+    public ResponseEntity postMessage(
+            @CurrentUser AuthenticatedUser authenticatedUser,
+            @PathVariable UUID chatId,
+            @RequestBody CreateMessageDto createMessageDto) {
+        log.info("Creating new message for chat {} for user {}.", chatId, authenticatedUser.getUserId());
+        chatService.createMessage(chatId, authenticatedUser.getUserId(), createMessageDto);
+        return new ResponseEntity(HttpStatus.CREATED);
+    }
+
+    @ExceptionHandler({BusinessException.class})
+    public ResponseEntity handle(Throwable throwable) {
+        return ResponseEntity.badRequest().build();
+    }
```

Comment:
I haven't added the polling endpoints yet. We could of course also use the /chats endpoint for polling, and if the user is currently in a chat the /chats/id/messages endpoint. Then we need no separate polling endpoint for the MVP.
After the MVP we could then think about websockets or server sent events. With this we could start simple (when we do not have multiple instances), but when we want to scale it gets complicated, as the clients are connected to one instance and do not get updated for updates on other instances. This could be solved with a redis cluster where dancer subscribes on so that every instance gets every update.
With SSE we could still use our POST endpoints from the frontend and only use SSE to update the clients. In general SSE is unidirectional but easier to set up (also HTTP based). 
With websockets we would not need our POST endpoints anymore as websockets works bidirectional, but of course we could still use them and only provide updates from the backend via websockets.